### PR TITLE
added all main functionality

### PR DESCRIPTION
**Pull request by katamoritoka**

### CODE REVIEW

File: components/TodoList.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+          <Todo
+            text={item.text}
+            done={item.done}
+            id={item.id}
+            delete={deleteTodo}
+            toggle={toggleTodoStatus}
+            edit={edit}
+          />
+        )}
+        keyExtractor={(item, index) => "" + item.id}
```
Comment (oeswww) [inline components/TodoList.js]: @katamoritoka `#{item.id}` or `item.id.toString()`

File: components/TodoList.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        </TouchableHighlight>
+      </View>
+      <Text style={styles.counter}>
+        Uncompleted: {uncompletedTodosCount()}, total: {todos.length}
+      </Text>
+      <FlatList
+        data={todos}
+        renderItem={({ item }, index) => (
+          <Todo
+            text={item.text}
```
Comment (oeswww) [inline components/TodoList.js]: @cray0000 {...item}

File: components/Todo.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  } else {
+    return (
+      <TouchableHighlight onLongPress={handleLongPress}>
+        <View style={[styles.container, styles.todo]}>
+          <TouchableHighlight
+            onPress={handleCheckButtonPress}
+            style={[
+              styles.button,
+              props.done
+                ? { backgroundColor: "lightgreen" }
```
Comment (oeswww) [inline components/Todo.js]: @katamoritoka please don't write inline styles (everywhere)

File: components/App.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import React from "react";
+import { Text, View } from "react-native";
+import styles from "./styles";
+import TodoList from "./TodoList";
+
+export default function App() {
```
[the remark this answers is not in the document]
Comment (katamoritoka) [reply components/App.js]: done

File: components/TodoList.js [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+  View,
+  TouchableHighlight,
+  FlatList,
+  TextInput
+} from "react-native";
+import Todo from "./Todo";
+import styles from "./styles";
+
+export default function TodoList(props) {
+  const [counter, setCounter] = useState(0);
```
[the remark this answers is not in the document]
Comment (katamoritoka) [reply components/TodoList.js]: done

File: components/TodoList.js [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+  const [todos, setTodos] = useState([]);
+  const [input, setInput] = useState("");
+
+  function handleAddButtonPress() {
+    if (input !== "") {
```
[the remark this answers is not in the document]
Comment (katamoritoka) [reply components/TodoList.js]: done

File: components/TodoList.js [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+      addTodo(input);
+      setInput("");
+    }
+  }
+
+  function addTodo(text) {
+    let currentTodos = todos.map(todo => todo);
+    currentTodos.unshift({ text: text, done: false, id: counter });
```
[the remark this answers is not in the document]
Comment (katamoritoka) [reply components/TodoList.js]: done

File: components/TodoList.js [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+    setTodos(currentTodos);
+  }
+
+  function handleInputChange(text) {
+    setInput(text);
+  }
+
+  function uncompletedTodosCount() {
+    const uncompleted = todos.filter(todo => !todo.done);
+    return uncompleted.length;
```
[the remark this answers is not in the document]
Comment (katamoritoka) [reply components/TodoList.js]: done

File: components/TodoList.js [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+  }
+
+  function edit(id, newText) {
+    let currentTodos = todos.map(todo => todo);
```
[the remark this answers is not in the document]
Comment (katamoritoka) [reply components/TodoList.js]: done

File: components/TodoList.js (hunk 1 above)
Comment (katamoritoka) [reply components/TodoList.js]: done

File: components/TodoList.js (hunk 2 above)
Comment (katamoritoka) [reply components/TodoList.js]: done

File: components/TodoList.js [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+
+export default function TodoList(props) {
+  const [counter, setCounter] = useState(0);
+  const [todos, setTodos] = useState([]);
+  const [input, setInput] = useState("");
+
+  function handleAddButtonPress() {
+    if (input !== "") {
+      addTodo(input);
+      setInput("");
```
[the remark this answers is not in the document]
Comment (katamoritoka) [reply components/TodoList.js]: done

File: components/Todo.js (hunk 3 above)
Comment (katamoritoka) [reply components/Todo.js]: done

File: components/TodoList.js [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+    setTodos(currentTodos);
+  }
+
+  function deleteTodo(id) {
+    let currentTodos = todos.map(todo => todo);
+    const index = currentTodos.findIndex(todo => todo.id === id);
+    let start = currentTodos.slice(0, index);
+    let end = currentTodos.slice(index + 1);
+    currentTodos = start.concat(end);
+    setTodos(currentTodos);
```
[the remark this answers is not in the document]
Comment (katamoritoka) [reply components/TodoList.js]: done

File: components/TodoList.js [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+  }
+
+  function toggleTodoStatus(id) {
+    let currentTodos = todos.map(todo => todo);
```
[the remark this answers is not in the document]
Comment (katamoritoka) [reply components/TodoList.js]: done

File: components/TodoList.js [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+  }
+
+  return (
+    <View style={styles.container}>
+      <View style={styles.todo}>
+        <TextInput
+          selectTextOnFocus={true}
+          style={styles.text}
+          placeholder="Write new todo here"
+          onChangeText={handleInputChange}
```
[the remark this answers is not in the document]
Comment (katamoritoka) [reply components/TodoList.js]: done

**Pull request merged**